File: Examples/test-suite/javascript/catches_strings_runme.js
var catches_strings = require("catches_strings");

var exception_thrown = false;
try {
  /* await */(catches_strings.StringsThrower.charstring());
} catch (e) {
  if (!(e.message || e).includes("charstring message"))
    throw new Error("incorrect exception message " + e.message);
  exception_thrown = true;
}
if (!exception_thrown)
  throw new Error("Should have thrown an exception");

exception_thrown = false;
try {
  /* await */(catches_strings.StringsThrower.stdstring());
} catch (e) {
  if (!(e.message || e).includes("stdstring message"))
    throw new Error("incorrect exception message " + e.message);
  exception_thrown = true;
}
if (!exception_thrown)
  throw new Error("Should have thrown an exception");

File: Examples/test-suite/csharp/catches_strings_runme.cs
using System;
using catches_stringsNamespace;

public class catches_strings_runme {
  public static void Main()
  {
    {
      bool exception_thrown = false;
      try {
        StringsThrower.charstring();
      } catch (ApplicationException e) {
        if (!e.Message.Contains("charstring message"))
          throw new ApplicationException("incorrect exception message:" + e);
        exception_thrown = true;
      }
      if (!exception_thrown)
        throw new ApplicationException("Should have thrown an exception");
    }
    {
      bool exception_thrown = false;
      try {
        StringsThrower.stdstring();
      } catch (ApplicationException e) {
        if (!e.Message.Contains("stdstring message"))
          throw new ApplicationException("incorrect exception message:" + e);
        exception_thrown = true;
      }
      if (!exception_thrown)
        throw new ApplicationException("Should have thrown an exception");
    }
  }
}
